dallinger/heroku/messages.py

```diff
@@ -147,102 +154,225 @@ def get_email_server():
 """
 
 
-class EmailingHITMessager(object):
+class MessengerError(Exception):
+    """A message could not be relayed."""
+
+
+class HITSummary(object):
 
-    def __init__(self, when, assignment_id, hit_duration, time_active, config,
-                 server=None, app_id=None):
+    def __init__(self, assignment_id, duration, time_active, app_id, when=datetime.now()):
         self.when = when
         self.assignment_id = assignment_id
-        self.duration = int(round(hit_duration / 60))
+        self.duration = int(round(duration / 60))
         self.minutes_so_far = int(round(time_active / 60))
-        self.minutes_excess = int(round((time_active - hit_duration) / 60))
-        self.whimsical = config.get("whimsical")
-        self.username = config.get('dallinger_email_username')
-        self.fromaddr = self.username + "@gmail.com"
-        self.toaddr = config.get('contact_email_on_error')
-        self.email_password = config.get("dallinger_email_key")
-        self.server = server or get_email_server()
+        self.minutes_excess = int(round((time_active - duration) / 60))
         self.app_id = app_id
 
-    def _send(self, data):
-        msg = MIMEText(data['message'])
-        msg['Subject'] = data['subject']
-        self.server.starttls()
-        self.server.login(self.username, self.email_password)
-        self.server.sendmail(self.fromaddr, self.toaddr, msg.as_string())
-        self.server.quit()
+
+CONFIG_PLACEHOLDER = '???'
+
+
+class EmailConfig(object):
+    """Extracts and validates email-related values from a Configuration
+    """
+    _map = {
+        'username': 'smtp_username',
+        'toaddr': 'contact_email_on_error',
+        'fromaddr': 'dallinger_email_address',
+        # 'email_password': 'smtp_password',
+    }
+
+    def __init__(self, config):
+        self.host = config.get('smtp_host', '')
```

Comment:
Should we retain the gmail default for backwards compatibility?

Reply:
It's now a default in global_config_defaults.txt, which felt like the right place to me, since it's more in the "user space" and has documentation in configuration.rst.